synthesizer/program/src/logic/command/metadata_get.rs

```diff
@@ -0,0 +1,303 @@
+// Copyright 2024 Aleo Network Foundation
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    Opcode,
+    traits::{FinalizeStoreTrait, RegistersLoad, RegistersStore, StackMatches, StackProgram},
+};
+use console::{
+    network::prelude::*,
+    program::{Identifier, Literal, LiteralType, Plaintext, PlaintextType, ProgramID, Register, Value},
+};
+
+/// The name of the metadata.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum MetadataName<N: Network> {
+    /// The program checksum.
+    Checksum,
+    /// A declared metadata identifier.
+    Identifier(Identifier<N>),
+}
+
+/// A command to get metadata about a program, e.g. `metadata.get program_owner into r1 as address;`.
+/// Gets the value with the `name` from the program and stores it in the `destination` register.
+/// The value is checked to be of the `destination_type`.
+///
+/// `metadata.get _checksum into r1 as u128;` is a special case where the metadata is not retrieved from the program.
```

Comment:
Given that the checksum behaves quite differently, did you consider making a separate checksum command? What is the biggest downside, more lines of code?

Reply:
Yeah exactly. But it could be split out fairly easily.